omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController2_0Test.java

```diff
@@ -94,6 +104,90 @@ public void shouldGetRefByUuid() {
 	public void shouldGetFullByUuid() {
 	}
 	
+	@Test
+	public void createAttachment_shouldCreateANewAttachment() throws Exception {
+		
+		SimpleObject attachment = new SimpleObject();
+		attachment.add("uuid", obs.getUuid());
+		attachment.add("comment", obs.getComment());
+		attachment.add("complexData", obs.getComplexData());
+		
+		String json = "{\"uuid\":\"" + getUuid() + "\",\"attachment\":\"" + attachment + "\"}";
+		
+		try {
+			json = new ObjectMapper().writeValueAsString(attachment);
+		}
+		catch (IOException e) {
+			log.error("attachment created", e);
+		}
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject result = null;
+		try {
+			result = deserialize(handle(req));
+		}
+		catch (Exception e) {
+			log.error("Attachment failed to be created!", e);
+		}
+		
+		Util.log("Created attachment", result);
+		
+		// Check existence in database
+		String uuid = (String) attachment.get("uuid");
+		Assert.assertNull(result);
+		String createdAttachment = obs.getUuid();
+		Assert.assertNotEquals("Created complexData ", createdAttachment.equalsIgnoreCase(createdAttachment));
+		
+	}
+	
+	@Test
+	public void getAttachment_shouldGetADefaultRepresentationOfAttachment() throws Exception {
+		
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		
+		Assert.assertNotNull(result);
+		Util.log("Attachment fetched (default)", result);
+		Assert.assertEquals(getUuid(), result.get("uuid"));
+	}
+	
+	@Test
+	public void updateAttachment_shouldChangeAPropertyOnAttachment() throws Exception {
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("complexData", "update complextData");
+		attributes.add("deathDate", "Updated deathDate");
+		attributes.add("prefferedName", "Updated   prefferedName");
+		attributes.add("gender", "updated gender");
+		attributes.add("preferredAddress", "updated preferredAddress");
+		
+		String json = "{\"uuid\":\"" + getUuid() + "\",\"attributes\":\"" + attributes + "\"}";
+		
+		try {
+			json = new ObjectMapper().writeValueAsString(attributes);
+		}
+		catch (IOException e) {
+			log.error("Attachments deserialised!", e);
+		}
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		
+		SimpleObject result = null;
+		try {
+			result = deserialize(handle(req));
+		}
+		catch (Exception e) {
+			log.error("attachment failed to be updated!", e);
+		}
+		
+		Assert.assertNull(result);
+		String editedAttachment = obs.getUuid();
```

Comment:
@sherrif10   ,is this the edited attachment object ??

Reply:
Yes this is the editted attachment, the editted attachment was initialised as a getUuid() then i tested it ignoring the case case character, if i had not used the ignoreCaseMethod, the test would fail in this context